### PR TITLE
Serverless Function/API setup

### PR DESCRIPTION
**Pull request by WillDera**
This PR is the setup for our serverless API with nest.js. 

**### ToDos**

- [x] Set up `/nonce` endpoint which receives the user's "wallet address" and returns `byte32 hash` to be signed by the user with their wallet.
- [x] Set up `/signature/{signature}` endpoint which receives the `signature` from the backend (R-on-R API) and `recovers` the user's wallet address from the signature, if successful would return `true` else `false`.

### CODE REVIEW

File: apps/mint-check/pages/api/signature.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+   * @param Req The payload containing Address, Hash and Signature
+   * @returns true if address is successfully recovered from signature @else false
+   * 
+*/ 
+async function GenerateNonce (req: NextApiRequest, res: NextApiResponse) {
+  const { walletAddress, nonce, signature } = req.body
+
+  const recoveredAddress =  ethers.utils.verifyMessage(nonce.message, signature)
+  
+  const bool = (recoveredAddress === walletAddress)
```
Comment (carlomigueldy) [inline apps/mint-check/pages/api/signature.ts:20]: Btw add this just to be sure it is always matching
```ts
recoveredAddress.toLowerCase() === walletAddress.toLowerCase()
```

**Review by carlomigueldy — CHANGES_REQUESTED** [review]
Nice one brother @WillDera 

1 very minor change, is just to add `.toLowerCase()` to make sure they always match

File: apps/mint-check/pages/api/signature.ts (hunk 1 above)
Comment (WillDera) [reply apps/mint-check/pages/api/signature.ts:20]: Done

**Review by carlomigueldy — APPROVED** [review]
LGTM!

**Review by Kzoeps — APPROVED** [review]

**Pull request merged**